### PR TITLE
Snmp: Extend test coverage #4797

### PR DESCRIPTION
**Pull request by JiriOndrusek**
fixes https://github.com/apache/camel-quarkus/issues/4797

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: integration-tests-jvm/snmp/src/test/java/org/apache/camel/quarkus/component/snmp/it/SnmpTest.java [hunk 1]
```diff
@@ -16,19 +16,80 @@
...
+
+    @Test
+    public void testGetNext() {
+
+        RestAssured.given()
+                .body("TEXT")
+                .post("/snmp/getNext")
+                .then()
+                .statusCode(200)
+                .body(Matchers.equalTo("Response from the test #1,Response from the test #2"));
```
Comment (jamesnetherton) [inline integration-tests-jvm/snmp/src/test/java/org/apache/camel/quarkus/component/snmp/it/SnmpTest.java:93]: Does this test depend on `testProducePDU` being executed first?

Comment (JiriOndrusek) [reply integration-tests-jvm/snmp/src/test/java/org/apache/camel/quarkus/component/snmp/it/SnmpTest.java:93]: No, tests uses different OID and the test resource returns for each of oids a freshly counted responses.

File: integration-tests-jvm/snmp/src/test/java/org/apache/camel/quarkus/component/snmp/it/SnmpTestResource.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+        public TestCommandResponder(Snmp commandResponder) {
+            this.commandResponder = commandResponder;
+        }
+
+        @Override
+        public synchronized void processPdu(CommandResponderEvent event) {
+            PDU pdu = event.getPDU();
+            Vector<? extends VariableBinding> vbs = Optional.ofNullable(pdu.getVariableBindings()).orElse(new Vector<>(0));
+            String key = vbs.stream().sequential().map(vb -> vb.getOid().toString()).collect(Collectors.joining(","));
```
Comment (JiriOndrusek) [inline integration-tests-jvm/snmp/src/test/java/org/apache/camel/quarkus/component/snmp/it/SnmpTestResource.java:91]: @jamesnetherton  there is a differencerfor  tests based on variable bindings

File: integration-tests-jvm/snmp/src/test/java/org/apache/camel/quarkus/component/snmp/it/SnmpTest.java [hunk 3]
```diff
@@ -16,19 +16,80 @@
...
 
+import static org.awaitility.Awaitility.await;
+
+/**
+ * There is a responder defined in the test resource. Which returns 2 responses without a delay and the third one
+ * with delay longer then default timeout. This means following behavior:
+ * - send PDU will receive 1 response
+ * - get_next will receive 2 responses (the third one reaches timeout)
+ * - poll returns unending stream of responses
+ */
```
Comment (JiriOndrusek) [inline integration-tests-jvm/snmp/src/test/java/org/apache/camel/quarkus/component/snmp/it/SnmpTest.java:35]: @jamesnetherton here is a simple explanation, why different test cases asserts different values

**Review by jamesnetherton — APPROVED** [review]

**Pull request merged**